leader-election-impl/src/main/java/com/palantir/leader/LeadershipEvents.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.paxos.PaxosRoundFailureException;
+import com.palantir.paxos.PaxosValue;
+
+class LeadershipEvents {
+
+    private static final String LEADER_LOG_NAME = "leadership";
+    private static final Logger leaderLog = LoggerFactory.getLogger(LEADER_LOG_NAME);
+
+    private final Meter gainedLeadership;
+    private final Meter lostLeadership;
+    private final Meter noQuorum;
+    private final Meter proposedLeadership;
+    private final Meter proposalFailure;
+
+    public LeadershipEvents(MetricRegistry metrics) {
+        gainedLeadership = metrics.meter("leadership.gained");
+        lostLeadership = metrics.meter("leadership.lost");
+        noQuorum = metrics.meter("leadership.no-quorum");
+        proposedLeadership = metrics.meter("leadership.proposed");
+        proposalFailure = metrics.meter("leadership.proposed.failure");
+    }
+
+    public void proposedLeadershipFor(long round) {
+        leaderLog.info("Proposing leadership", SafeArg.of("round", round));
+        proposedLeadership.mark();
+    }
+
+    public void gainedLeadershipFor(PaxosValue value) {
+        leaderLog.info("Gained leadership", SafeArg.of("value", value));
+        gainedLeadership.mark();
+    }
+
+    public void lostLeadershipFor(PaxosValue value) {
+        leaderLog.info("Lost leadership", SafeArg.of("value", value));
```

Comment:
Yep, sorry about that - we'd be looking for the value that replaced us. In that case I don't think it's as valuable (finding what was lost is a lot trickier than finding who became the leader for the new term).